ENH: allow image to interpolate post RGBA (#18782)
matplotlib/matplotlib: doc/users/next_whats_new/image_interpolation_rgbastage.rst

Image interpolation now possible at RGBA stage
----------------------------------------------

Images in Matplotlib created via `~.axes.Axes.imshow` are resampled to match 
the resolution of the current canvas.  It is useful to apply an anto-aliasing
filter when downsampling to reduce Moire effects.  Previously this was 

> done at the data stage, so adjoining data values would be interpolated 
> between, and then the data would be passed to the normalization, and then 

Reviewer: 'interpolated between' is a bit awkward, due to the dangling preposition. I'm not fully happy with it, but maybe 'interpolated together'?

Could also drop 'and then the data would be'.
```suggestion
done at the data stage, so adjoining data values would be interpolated 
together, passed to the normalization, and finally to 
```

Author: Changed to 

```
By default, interpolation 
is done on the data, a norm applied, and then the colormapping performed. 
```

the colormap where it is now an array of RGBA values.

However, it is often desireable for the anti-aliasing interpolation to happen 
in RGBA space, where the colors are interpolated rather than the data.  This 

> usually leads to data outside the colormap, but visually blends adjacent 

Reviewer: ```suggestion
usually leads to colors outside the colormap, but visually blends adjacent 
```

colors, and is what browsers and other image processing software does. 

A new keyword argument *interpolation_stage* is provided for 
`~.axes.Axes.imshow` to set the stage at which the anti-aliasing interpolation 
happens.  The default is the current behaviour of "data", with the alternative

> being "rgba" for the new behavior.  

Reviewer: It's not (afaik) going to replace it as default:
```suggestion
being "rgba" for the newly-available behavior.  
```
